apps/files_sharing/lib/Controller/ShareAPIController.php

```diff
@@ -646,12 +646,16 @@ public function createShare(
 		}
 
 		//Expire date
-		if ($expireDate !== '') {
-			try {
-				$expireDateTime = $this->parseDate($expireDate);
-				$share->setExpirationDate($expireDateTime);
-			} catch (\Exception $e) {
-				throw new OCSNotFoundException($this->l->t('Invalid date, date format must be YYYY-MM-DD'));
+		if ($expireDate !== null) {
+			if ($expireDate !== '') {
+				try {
+					$expireDateTime = $this->parseDate($expireDate);
+					$share->setExpirationDate($expireDateTime);
+				} catch (\Exception $e) {
+					throw new OCSNotFoundException($this->l->t('Invalid date, date format must be YYYY-MM-DD'));
+				}
+			} else {
+				$share->setExpirationDate('');
```

Comment:
```suggestion
				$share->setExpirationDate(null);
```

Since, `null` is the basis for setting default expiration date in the share manager for new shares, we just can't set this to null here.


But can be done much later in `validateExpirationDateInternal` and `validateExpirationDateLink`